### ISSUE TITLE
Remote debugging works from command line but not from VS Code

### ISSUE THREAD

**Issue by rpfister51** [Microsoft/vscode-go#1138]
I cannot debug a docker process from VS Code.  I am able to connect to the docker container from the delve command line, but not from VS Code.  From the command line I can set breakpoints, list breakpoints, and step through the code.  I cannot do any of these from VS Code.

The easiest way to reproduce this is:   

1. Clone the debugging branch of https://github.com/lukehoban/webapp-go/tree/debugging  
2. Change line 1 of Dockerfile from "FROM golang:1.6" to "FROM golang:latest" (The docker build failed for me when I used golang:1.6.)  
3. Change host to "127.0.0.1" (assuming you are running on the same machine).  
4. Follow instructions in the comments in the launch.json; i.e., run the commands "$ docker build -t webapp-go ." and "$ docker run -d --name webapp-go --privileged -p 8080:8080 -p 2345:2345 webapp-go" from the command prompt.
5. Set breakpoints. I used lines 37, 47 and 55 in main.go
6. Launch the "Remote debug in Docker" debug configuration.
7. To exercise the app use "http://localhost:8080/edit/foo".  (You can later try /view/foo and /save/foo and other files besides foo.)

Now the debug pause, restart and stop buttons are active, but step over, step into, and step out are not.   Breakpoints are greyed out circles, and hover message is Unverified Breakpoint.  See screenshot:

![webapp frozen](https://user-images.githubusercontent.com/12035948/29723689-8a77134e-8993-11e7-8afe-5a0872d67fdd.png)

Exercising the app from the browser triggers no breaks.  If, on the other hand, the breaks are set from the delve command line, exercising the app does cause it to stop at the breakpoints that are set.

Versions I am using:

go version go1.8.3 windows/amd64

docker version:
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:30:30 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:51:55 2017
 OS/Arch:      linux/amd64
 Experimental: true

Visual Studio Code 1.14.2

Go extension (lukehoban.go) 0.6.63, 26th July, 2017

Delve Version:
Delve Debugger
Version: 1.0.0-rc.1
Build: $Id: b6077a6a35c48a31716aad2ad3fdf77bbdf70abd $

**Comment by ilevine** [comment]
Getting the same problem - any news ?

**Comment by rpfister51** [comment]
Hi @ilevine, no other notice of this problem but yours.  I did figure out why the FROM golang:1.6 no longer works.  There was a breaking change which is discussed in issue 936 that is causing this problem.  Go 1.7 or higher is now required to support the current version of Delve.

**Comment by ramya-rao-a** [comment]
@ilevine Are you on Windows too?

There was an issue with mapping remote paths to local paths in Windows and the fix for it is in the master branch. Can you try out the extension from source following the below steps and let me know if the issue is resolved for you?

- Download https://github.com/Microsoft/vscode-go/blob/master/Go-latest.vsix
- Run `code --install-extension Go-latest.vsix` from the location where the above is downloaded
- Reload VS Code

**Comment by rpfister51** [comment]
Applying the change suggested by ramya solved this problem for me.  Remote debugging to a golang process within a docker container is now no longer frozen.  The icons for step over, step into and step out are now active after beginning a debug session.  I have successfully used the step over icon and been able to inspect state as I step through my code.

I did bump into a new problem though. When stepping into some code in the golang standard library the debugger threw an error.  The path to the golang source was not valid.  I'll close this issue and write the new problem as a separate issue.
